File: solution/2191. Sort the Jumbled Numbers/2191.cpp
class Solution {
public:
    vector<int> sortJumbled(vector<int>& mapping, vector<int>& nums) {
        const int n = nums.size();
        
        vector<pair<int, int>> A(n);
        for (int i = 0; i < n; ++i) {
            int hash = 0;
            string x = to_string(nums[i]);
            for (char& c : x)
                hash = hash * 10 + mapping[c - '0'];
            A[i] = {hash, i};
        }
        
        sort(A.begin(), A.end());
        
        vector<int> res(n);
        for (int i = 0; i < n; ++i)
            res[i] = nums[A[i].second];
        return res;
    }
};

File: solution/2191. Sort the Jumbled Numbers/2191.py
class Solution:
    def sortJumbled(self, mapping: List[int], nums: List[int]) -> List[int]:
        
        def hash(num):
            x = ''.join(str(mapping[int(digit)]) for digit in str(num))
            return int(x)
        
        A = [(hash(nums[i]), i) for i in range(len(nums))]
        A.sort()
        return [nums[i] for _, i in A]
    
